### ISSUE TITLE
InvalidJavadocPositionCheck: false positive on correctly-placed Javadoc in compact source files (JEP 512)

### ISSUE THREAD

**Issue by vivek-0509** [checkstyle/checkstyle#20217]
Parent Issue #19971

I have read check documentation: https://checkstyle.org/checks/javadoc/invalidjavadocposition.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > config.xml <<'EOF'
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="InvalidJavadocPosition"/>
  </module>
</module>
EOF

vivek@Viveks-MacBook-Air checkstyle % cat > InputInvalidJavadocPositionCompact.java <<'EOF'
/** This field docs */
int value = 42;

/** This method docs */
void main() { }
EOF

vivek@Viveks-MacBook-Air checkstyle % RUN_LOCALE="-Duser.language=en -Duser.country=US"
java $RUN_LOCALE -jar target/checkstyle-13.5.0-SNAPSHOT-all.jar -c config.xml InputInvalidJavadocPositionCompact.java

Starting audit...
[ERROR] /Users/vivek/checkstyle/InputInvalidJavadocPositionCompact.java:1:1: Javadoc comment is placed in the wrong location. [InvalidJavadocPosition]
[ERROR] /Users/vivek/checkstyle/InputInvalidJavadocPositionCompact.java:4:1: Javadoc comment is placed in the wrong location. [InvalidJavadocPosition]
Audit done.
Checkstyle ends with 2 errors.
```
---

**Describe what you expect in detail.**

InvalidJavadocPositionCheck must not flag correctly-placed Javadoc before top-level fields and methods in a JEP 512 compact source file as being in the wrong location. The check delegates to `JavadocUtil.isCorrectJavadocPosition`, which accepts a Javadoc only if it sits on a type, on a member, or on a package declaration; the "on a member" branch walks the parent chain looking for an `OBJBLOCK` ancestor. In a compact source file the implicit class has no `OBJBLOCK`, so the check rejects all top-level Javadoc as misplaced, even though the same Javadoc inside a normal class is correctly accepted.

Expected: no violation on Javadoc that immediately precedes a top-level field or method declaration in a compact source file, identical to how the same Javadoc is treated inside a normal class. Genuinely misplaced Javadoc (e.g. inside a method body before a statement) should still be flagged as today.

**Comment by vivek-0509** [comment]
To other contributors: please do not pick this issue for now to avoid duplicate work, as I may already be working on it. There are many other open issues available, so please consider picking those and help us improve the project. Thanks for understanding!

**Comment by romani** [comment]
Fix is merged
